File: ssqq.capacitor-onebot-connector/ios/Sources/OnebotPlugin/OnebotPlugin.swift
import Foundation
import Capacitor
import os

/**
 * https://capacitorjs.com/docs/plugins/ios
 */
@available(iOS 14.0, *)
@objc(OnebotPlugin)
public class OnebotPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "OnebotPlugin"
    public let jsName = "Onebot"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "connect", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "close", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "send", returnType: CAPPluginReturnPromise),

        CAPPluginMethod(name: "findService", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = Onebot()
    private let logger = Logger()

    override public func load() {
        super.load()
        logger.info(" _____ _____ _____ _____ __ __  ")
        logger.info("|   __|_   _|  _  |  _  |  |  | ")
        logger.info("|__   | | | |     |   __|-   -| ")
        logger.info("|_____| |_| |__|__|__|  |__|__| CopyRight © Stapx Steve")
        logger.info("=======================================================")
        logger.info("Capacitor Onebot 基础连接器加载完成")
        logger.info("当前执行平台：iOS / iPadOS")
        logger.info("** 此插件 swift 版本代码 95% 由 Github Copilot 及 OpenAI ChatGPT 4o 生成 **")

        // 注册 implementation 的 onEvent 回调，需要判断下有没有注册过
        if implementation.onEvent == nil {
            implementation.onEvent = { type, data in
                self.notifyListeners("onebot:event", data: [ "type": type, "data": data ])
            }
        }
    }

    @objc func connect(_ call: CAPPluginCall) {
        let url = call.getString("url") ?? ""
        call.resolve([
            "value": implementation.connect(url)
        ])
    }

    @objc func close(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.close()
        ])
    }

    @objc func send(_ call: CAPPluginCall) {
        let data = call.getString("data") ?? ""
        call.resolve([
            "value": implementation.send(data)
        ])
    }

    @objc func findService(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.findService()
        ])
    }

}

File: ssqq.capacitor-onebot-connector/android/src/main/java/cn/stapxs/qqweb/connector/OnebotPlugin.java
package cn.stapxs.qqweb.connector;

import android.util.Log;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Onebot")
public class OnebotPlugin extends Plugin {

    private WebSocketClient webSocketClient;
    private final Onebot implementation = new Onebot();

    public void sendNotify(String type, String data) {
        JSObject ret = new JSObject();
        ret.put("type", type);
        ret.put("data", data);
        this.notifyListeners("onebot:event", ret);
    }

    @Override
    public void load() {
        super.load();
        Log.i("Plugin", " _____ _____ _____ _____ __ __  \n" +
                        "|   __|_   _|  _  |  _  |  |  | \n" +
                        "|__   | | | |     |   __|-   -| \n" +
                        "|_____| |_| |__|__|__|  |__|__| CopyRight © Stapx Steve\n" +
                        "=======================================================\n" +
                        "Capacitor Onebot 基础连接器加载完成");
        Log.i("Plugin", "当前执行平台：Android");
        Log.i("Plugin", "** 此插件 Java 版本代码 70% 由 Github Copilot 及 OpenAI ChatGPT 4o 生成 **");
    }

    @PluginMethod
    public void connect(PluginCall call) {
        String value = call.getString("url");

        JSObject ret = new JSObject();
        webSocketClient = implementation.connect(this, webSocketClient, value);
        ret.put("value", true);
        call.resolve(ret);
    }

    @PluginMethod
    public void send(PluginCall call) {
        String value = call.getString("data");

        JSObject ret = new JSObject();
        ret.put("value", implementation.send(webSocketClient, value));
        call.resolve(ret);
    }

    @PluginMethod
    public void close(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", implementation.close(webSocketClient));
        call.resolve(ret);
    }

    @PluginMethod
    public void findService(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", implementation.findService(this));
        call.resolve(ret);
    }
}
